generator/abstractmetabuilder.cpp

```diff
@@ -1962,9 +1967,9 @@ AbstractMetaType *AbstractMetaBuilder::inheritTemplateType(const QList<AbstractM
     }
 
     if (returned->hasInstantiations()) {
-        QList<AbstractMetaType *> instantiations = returned->instantiations();
+        auto instantiations = returned->instantiations();
         for (int i=0; i<instantiations.count(); ++i) {
-            instantiations[i] = inheritTemplateType(template_types, instantiations.at(i), ok);
+            instantiations[i] = AbstractMetaType::shared_pointer(inheritTemplateType(template_types, instantiations[i].data(), ok));
```

Comment:
Yes, the code would become cleaner, but eventually attempts to make code cleaner lead to the replacement `AbstractMetaType*` in all places with `AbstractMetaType::shared_pointer`. Also, since `type()` will return `shared_pointer`, `copy()` seems to be no longer needed: correct type destruction will be ensured. If you agree with these changes, I will make them.

Reply:
From my POV that would be ok - I guess @usiems will only be available next year, so I think you can go ahead...

Reply:
Since the copy method already existed, and the code originally didn't care much about memory leaks, one should check if the copy at the other places wasn't done because the type might subsequently be altered in the copy.